sensor/common/compliance/node_inventory_handler_impl.go

```diff
@@ -77,15 +77,24 @@ func (c *nodeInventoryHandlerImpl) run() <-chan *central.MsgFromSensor {
 					c.stopper.Flow().StopWithError(errInputChanClosed)
 					return
 				}
-				// TODO(ROX-12943): Do something with the inventory, e.g., attach NodeID
-				c.sendInventory(toC, inventory)
+				if inventory == nil {
+					log.Warnf("Received nil NodeInventory - not sending node inventory to Central")
+					break
+				}
+				if nodeID, err := c.nodeMatcher.GetNodeID(inventory.GetNodeName()); err != nil {
+					log.Warnf("Node '%s' unknown to sensor - not sending node inventory to Central", inventory.GetNodeName())
```

Comment:
I think we should add a test for this path. What would be expected if we received a `nil` inventory? What would happen if we don't find the node in the store?

Reply:
Not finding the Node in the store should by design result in ignoring the `nodeInventory` - spec in the ticket https://issues.redhat.com/browse/ROX-12943.

Reply:
Now if `nodeInvetory` is nil - this should not happen.

NodeInventories are written into the channel by [compliance service](https://github.com/stackrox/stackrox/blob/master/sensor/common/compliance/service_impl.go#L134-L151). The way the `switch` is built should prevent sending nils to the channel.

I understand that you ask for a testcase to see what happens if we get a nil here anyway. Let me see what I can implement in `sensor/common/compliance/node_inventory_handler_test.go`

Reply:
Got the test cases in [48c2883](https://github.com/stackrox/stackrox/pull/4484/commits/48c2883fe25debdeb89ebbc480230f8c7da300df) and [91c1ac1](https://github.com/stackrox/stackrox/pull/4484/commits/91c1ac188e3dc78adfd787e18a2b2c6519db5363) :)

Reply:
Thank you for adding the tests 🙂 

Sorry, I think my wording wasn't the best. My comment was simply pointing out that since this is the critical change in this PR, I think it should be tested to make sure it works the way it's meant to work.